### ISSUE TITLE
Weather not showing

### ISSUE THREAD

**Issue by FoxyBit** [Zren/plasma-applet-eventcalendar#86]
This has to be one of my favourite applets. The integration into the system and with Google calendar is just amazing and very useful.

But after upgrading to Plasma 5.16.1, the weather is no longer showing (it was with the previous version of Plasma). I have not changed any of the settings. It is currently set to OpenWeatherMap, but this is greyed out so I am unable to try any other services (can't remember if this was the same before).

Is there anyway I can get the weather to show again? Or is this something to do with the compatibility with Plamsa 5.16.1?

Obviously this is not a big issue that stops the applet working, but it would be nice to have back.

**Comment by Zren** [comment]
This is related to issue #85 as yesterday the weather quota was hit as well. I'm not sure how long it'll be disabled for.

I logged in (which I haven't done in ages) and noticed their privacy policy updated as well. It looks like their a UK company that needs to comply with GDPR so that's probably why there's an update.

* https://openweather.co.uk/privacy-policy
* https://openweather.co.uk/storage/app/media/Terms/Openweather_website_terms_and_conditions_of_use.pdf
* https://openweather.co.uk/storage/app/media/Terms/Openweather_terms_and_conditions_of_sale.pdf

> 6 You agree that you will not take any action that imposes an unreasonable or disproportionately large load on the infrastructure of the Site or our systems or networks, or any systems or networks connected to the Site.

Well, it wasn't a deliberate action. Definitely gotta throttle the weather requests in a better fashion. And disconnect a manual update of the events from updating the weather.

Gonna finish reading all that stuff before I can see the other API admin tools to give you a firm date on when weather will be back. I might subscripte to a paid tier for a few months as an apology for hitting their service so much this past week.

**Comment by Zren** [comment]
I've no idea if we were temporarily blocked before yesterday. OWM does not have a metrics/quota graph like Google does. They do have a log for when the API key is blocked and unblocked however.

![2019-06-20___12-44-19](https://user-images.githubusercontent.com/416367/59866251-28f49200-9359-11e9-8126-3499b058547b.png)

So 90000 seconds is 25 hours. It was blocked on June 19th at 8pm UTC. So it should be unblocked today June 20th at 9pm UTC. That converts to 5pm EST.

**Comment by FoxyBit** [comment]
Yes, API requests really get you. I have my own OWM API that I use for a couple of test websites, so maybe you could add a feature that would allow people to use their own API? Leaving those that don't want to and/or don't know how to use an API to have the weather displayed as normal. This may decrease your usage a lot.
